lib/pf/role.pm

```diff
@@ -395,14 +395,55 @@ sub getRegisteredRole {
     if( $role ) {
         return $role;
     }
-
     # Try MAC_AUTH, then other EAP methods and finally anything else.
-    if ( $args->{'connection_type'} && ($args->{'connection_type'} & $WIRED_MAC_AUTH) == $WIRED_MAC_AUTH ) {
-        $logger->info("Connection type is Ethernet-NoEAP. Getting role from node_info" );
-        $role = $args->{'node_info'}->{'category'};
-    } elsif ( $args->{'connection_type'} && ($args->{'connection_type'} & $WIRELESS_MAC_AUTH) == $WIRELESS_MAC_AUTH ) {
-        $logger->info("Connection type is Wireless-802.11-NoEAP. Getting role from node_info" );
-        $role = $args->{'node_info'}->{'category'};
+    if ( $args->{'connection_type'} && ( ( ($args->{'connection_type'} & $WIRED_MAC_AUTH) == $WIRED_MAC_AUTH ) || ( ($args->{'connection_type'} & $WIRELESS_MAC_AUTH) == $WIRELESS_MAC_AUTH ) ) ) {
+        $logger->info("Connection type is MAC-AUTH. Getting role from node_info" );
+        my @sources = $profile->getFilteredAuthenticationSources($args->{'stripped_user_name'}, $args->{'realm'});
+        my @source = grep {uc($_->{'type'}) eq "AUTHORIZATION"} @sources;
+        if (@source) {
+            my $stripped_user = '';
+            $stripped_user = $args->{'stripped_user_name'} if(defined($args->{'stripped_user_name'}));
+            my $params = {
```

Comment:
Can we do a subroutine for building params from args that we could reuse elsewhere in this module?